src/views/tutorial-view/utils/variants/index.ts

```diff
@@ -25,6 +26,19 @@ export function getVariantParam(
 	return `${slug}:${optionSlug}`
 }
 
+export function handleVariantCookie(id: string, optionId: string) {
+	const key = 'variants'
+	const allVariants = Cookies.get(key)
+	const variantsObj = allVariants ? JSON.parse(allVariants) : {}
```

Comment:
It might be a good idea to wrap the `JSON.parse()` in a try/catch, just to guard against any unexpected behavior if for some reason the cookie has been modified by something else. If it fails to parse we would just assign to `{}` in that case.